src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -212,11 +224,101 @@ public void close() throws IOException
               String.format("Expected [%,d] bytes, only saw [%,d], potential corruption?", size, bytesWritten)
           );
         }
+
+        if(!writerCurrentlyInUse)
+        {
+        	mergeWithSmoosher();
+        }
       }
     };
   }
+  
+  public OutputStream makeOutputStream(String filename) throws IOException
+  {
+    File retFile = createdFiles.get(filename);
+    if (retFile == null) {
+      retFile =  new File(baseDir, filename);
+      createdFiles.put(filename, retFile);
+      return new BufferedOutputStream(new FileOutputStream(retFile));
+    } else  {
+      return new BufferedOutputStream(new FileOutputStream(retFile));
+    } 
+  }
+  
+  public InputStream makeInputStream(String filename) throws IOException
+  {
+    final File retFile = createdFiles.get(filename);
+    return retFile == null ? null : new FileInputStream(retFile);
+  }
+  
+  private void mergeWithSmoosher() throws IOException
+  {
+	  //get processed elements from the stack and write.
+	 ArrayDeque<File> fileToProcess = files.clone();
+	 files = new ArrayDeque<>();
+	  while(! fileToProcess.isEmpty())
+	  {
+		  File file = fileToProcess.pop();
+		  add(file);
+		  file.delete();
+	  }
+  }
+  
+  private SmooshedWriter delegateSmooshedWriter(final String name,final long size) throws IOException
+  {	
+		return new SmooshedWriter () {
+			private int currOffset = 0;
+			private boolean open = true;
+			OutputStream writer= makeOutputStream(name);
```

Comment:
I don't think you need another method for `makeOutputStream()`.  Before returning the object you can create a File and then close over that with the `SmooshedWriter` object.